### PR TITLE
Upgrade module to 1.18, upgrade dependencies, fix format string error

### PR DESCRIPTION
**Pull request by jonathansharman**
This upgrades the `go.mod` to 1.18, updates and vendors dependencies, and fixes a small `Errorf` format string bug.

I believe the dependency update is required for this tool to continue working on go 1.18.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -483,7 +483,7 @@ func getMinorUpdateVersion(path string) (string, error) {
 	result := results[0]
 
 	if result.Error != nil {
-		return "", fmt.Errorf("error getting module info: %s", path, result.Error.Err)
+		return "", fmt.Errorf("error getting module info: %s", result.Error.Err)
```
Comment (jonathansharman) [inline main.go:486]: This is now a hard error. I just removed the extra parameter - not sure if you'd prefer to work that parameter into the error.

Comment (nathanjcochran) [reply main.go:486]: I will probably try to work that parameter into the error, but I can do that post-merge. Thanks!!

**Comment by nathanjcochran** [comment]
Thanks for the updates @jonathansharman!! Much appreciated :rocket:

File: main.go (hunk 1 above)
Comment (nathanjcochran) [reply main.go:486]: Actually, nevermind. It looks like the `path` is already being added to the error message after the error is returned from `getMinorUpdateVersion`, when it's wrapped again [here](https://github.com/nathanjcochran/upgrade/blob/master/main.go#L428). I bet the `path` variable you removed was accidentally left-over from before I started wrapping the errors in a higher-level error like that. Thanks for fixing that!

**Pull request merged**